Unity.gitignore

```diff
@@ -39,3 +39,5 @@ sysinfo.txt
 # Crashlytics generated file
 Assets/StreamingAssets/crashlytics-build.properties
 
+#Do not ignore plugins
+!/**/*.dll
```

Comment:
This came up in #1603 and after a bit of discussion I think I was concerned about this being recursively applied to all paths - can this rule be made more specific?

Reply:
```suggestion
!/Assets/Plugins/**/*.dll
```

Reply:
I guess? I'd need some input from people who are more familiar with Unity at this point.

cc @ksmit799 @theDrake @1nfinite9 @owenblacker @FutureClemens @kilian442 @mariuz for feedback (peeps who have made changes to the template over the last year)

Reply:
DLLs in a Unity project are allowed to be placed within any "Plugins" directory or "Plugins" sub-directory. Also, the "Plugins" directory itself does not need to be placed directly in the Assets folder and can be placed in any directory. And there might be multiple plugins directories in your project.

Therefore, valid places are:
Assets\[0-n Directories]\Plugins\[0-n Directories]\plugin.dll

Examples from existing plugins, that are working in my projects (Unity 2017 & 2018):
Assets\Plugins\x86_64\plugin.dll
Assets\Plugins\WSA\PhoneSDK81\x86\plugin.dll
Assets\MyTests\Plugins\x86_64\Test.dll

I am not sure if a line like this makes sense at all for a unity gitignore file, but if you do it correctly, it at least does not do any harm.

Reply:
```suggestion
!/Assets/**/Plugins/**/*.dll
```

![capture](https://user-images.githubusercontent.com/1447176/50162369-55095a80-02c5-11e9-946f-00f740d8cacb.PNG)

Thanks @FutureClemens.

Reply:
I'm not sure if this makes sense within the scope of the global Unity gitignore, seeing as the current Unity gitignore doesn't ignore any dll files by itself. The fact that an external gitignore is impacting on Unity's makes this more of an edge case then something that should be applied globally.